pkg/controllers/gateway/gateway_controller.go

```diff
@@ -164,6 +188,15 @@ func (r *GatewayReconciler) reconcileGateway(ctx context.Context, previous gatew
 	allHosts := trafficAccessor.GetHosts()
 	managedHosts := []ManagedHost{}
 
+	// Set the annotations to sync the class name from the parameters
+	for cluster, downstreamClass := range params.GetDownstreamClasses() {
```

Comment:
yeah same here, I think for the downstream class, each instance would have the same class